### PR TITLE
recipe for tantivy-py

### PR DESCRIPTION
**Pull request by mediocretech**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mediocretech** [comment]
@conda-forge/help-python, ready for review!

File: recipes/tantivy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - make
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - {{ compiler('cxx') }}
+
+  host:
+    - python >=3.8
+    - maturin <=1.3.2
+    - pip
+    - rust
```
Comment (xhochy) [inline recipes/tantivy/meta.yaml:31]: If you need rust, you should be putting it as a compiler in build

File: recipes/tantivy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/tantivy-{{ version }}.tar.gz
+  sha256: dce07fa2910c94934aa3d96c91087936c24e4a5802d839625d67edc6d1c95e5c
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  missing_dso_whitelist:
+    - "*/libgcc*"
```
Comment (xhochy) [inline recipes/tantivy/meta.yaml:17]: This doesn't look right and coukf be related to rust in host

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by mediocretech** [comment]
Build modification pending https://github.com/conda-forge/staged-recipes/pull/27151

File: recipes/tantivy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - "*lib/libgcc_s.so.1"  # [linux]
+
+requirements:
+  build:
+    - cmake
+    - make
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - {{ compiler('cxx') }}
+    - rust <1.80
```
Comment (xhochy) [inline recipes/tantivy/meta.yaml:26]:
```suggestion
    - {{ compiler('rust') }} <1.80
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/tantivy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  name: {{ name|lower }}-py
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/tantivy-{{ version }}.tar.gz
+  sha256: dce07fa2910c94934aa3d96c91087936c24e4a5802d839625d67edc6d1c95e5c
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```
Comment (xhochy) [inline recipes/tantivy/meta.yaml:14]: You also should call `cargo-bundle-licenses` here

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**